server_admin/topics/identity-broker/social/gitlab.adoc

```diff
@@ -0,0 +1,32 @@
+
+==== GitLab
+
+There are a number of steps you have to complete to be able to login to GitLab.  First, go to the `Identity Providers` left menu item
+and select `GitLab` from the `Add provider` drop down list.  This will bring you to the `Add identity provider` page.
+
+.Add Identity Provider
+image:{project_images}/gitlab-add-identity-provider.png[]
+
+You can't click save yet, as you'll need to obtain a `Client ID` and `Client Secret` from GitLab.  One piece of data you'll need from this
+page is the `Redirect URI`.  You'll have to provide that to GitLab when you register {project_name} as a client there, so
+copy this URI to your clipboard.
+
+To enable login with GitLab you first have to register an application project in
+https://docs.gitlab.com/ee/integration/oauth_provider.html[GitLab as OAuth2 authentication service provider].
+
+NOTE: GitLab often changes the look and feel of application registration, so these directions might not always be up to date and the
+      configuration steps might be slightly different.
+
+
+.Add a New App
+image:images/gitlab-developer-applications.png[]
+
+You'll have to copy the `Redirect URI` from the {project_name} `Add Identity Provider` page and enter it into the
+`Authorization callback URL` field on the GitLab `Register a new OAuth application` page.  Once you've completed this
+page you will be brought to the application's management page.
+
+.GitLab App Page
+image:images/gitlab-app-page.png[]
+
+You will need to obtain the client ID and secret from this page so you can enter them into the {project_name} `Add identity provider` page.
+Go back to {project_name} and specify those items.
```

Comment:
Simpler review this time. Suggested rewrite:

==== GitLab

There are a number of steps you have to complete to be able to login to GitLab.

First, go to the `Identity Providers` left menu item and select `GitLab` from the `Add provider` drop down list. This will bring you to the `Add identity provider` page.

.Add Identity Provider
image:{project_images}/gitlab-add-identity-provider.png[]

Before you can click `Save`, you must obtain a `Client ID` and `Client Secret` from GitLab.

NOTE: You will the `Redirect URI` from this page in a later step, which you will provide to GitLab when you register {project_name} as a client there.

To enable login with GitLab you first have to register an application project in
https://docs.gitlab.com/ee/integration/oauth_provider.html[GitLab as OAuth2 authentication service provider].

NOTE: GitLab often changes the look and feel of application registration, so what you see on the GitLab site may differ. If in doubt, see the GitLab documentation.

.Add a New App
image:images/gitlab-developer-applications.png[]

Copy the `Redirect URI` from the {project_name} `Add Identity Provider` page and enter it into the `Authorization callback URL` field on the GitLab `Register a new OAuth application` page.

.GitLab App Page
image:images/gitlab-app-page.png[]

When you are done registering, click Save. This will open the application management page in GitLab. Find the client ID and secret from this page so you can enter them into the {project_name} `Add identity provider` page.

To finish, return to {project_name} and enter them. Click `Save`.